homeassistant/components/amazon_devices/switch.py

```diff
@@ -0,0 +1,78 @@
+"""Support for switches."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, Final
+
+from aioamazondevices.api import AmazonDevice
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import AmazonConfigEntry
+from .entity import AmazonEntity
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass(frozen=True, kw_only=True)
+class AmazonSwitchEntityDescription(SwitchEntityDescription):
+    """Amazon Devices switch entity description."""
+
+    is_on_fn: Callable[[AmazonDevice], bool]
+    subkey: str
+
+
+SWITCHES: Final = (
+    AmazonSwitchEntityDescription(
+        key="do_not_disturb",
+        subkey="AUDIO_PLAYER",
+        translation_key="do_not_disturb",
+        is_on_fn=lambda _device: _device.do_not_disturb,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: AmazonConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Amazon Devices switches based on a config entry."""
+
+    coordinator = entry.runtime_data
+
+    async_add_entities(
+        AmazonSwitchEntity(coordinator, serial_num, switch_desc)
+        for switch_desc in SWITCHES
+        for serial_num in coordinator.data
+        if switch_desc.subkey in coordinator.data[serial_num].capabilities
+    )
+
+
+class AmazonSwitchEntity(AmazonEntity, SwitchEntity):
+    """Switch device."""
+
+    entity_description: AmazonSwitchEntityDescription
+
+    async def _switch_set_state(self, state: bool) -> None:
+        """Set desired switch state."""
+        await self.coordinator.api.set_do_not_disturb(self.device, state)
+        self.coordinator.data[self.device.serial_number].do_not_disturb = state
+        self.async_write_ha_state()
```

Comment:
1. Should we refresh after swtiching state?
2. In any case, we should abstract this logic away in the entity description

Reply:
1. Refresh is every 30s so I think is a bit slow.
2. I applied the same logic we used in Comelit integration